plumbing/object/commit.go

```diff
@@ -145,12 +152,33 @@ func (c *Commit) Decode(o plumbing.EncodedObject) (err error) {
 	r := bufio.NewReader(reader)
 
 	var message bool
+	var gpgsig bool
 	for {
 		line, err := r.ReadBytes('\n')
 		if err != nil && err != io.EOF {
 			return err
 		}
 
+		if gpgsig {
+			// Check if it's the end of a PGP signature.
+			if bytes.Contains(line, []byte(endpgp)) {
+				c.GPGSignature += string(endpgp) + "\n"
+				gpgsig = false
+			} else {
+				// Trim the left padding.
+				line = bytes.TrimLeft(line, " ")
```

Comment:
Is there a reason we need to Trim the padding? I notice that we add padding back in when we format the Commit. Will this be necessary for handing the signature into another routine that does the verification of the signature?

Reply:
The padding is just git's way of presenting it. The public key block at https://www.gnupg.org/signature_key.html shows it without any left padding. So I thought that's how they are actually stored as a single signature. And yes, I thought it would be easier to perform signature verification in the future, without any need to trim the padding.